### ISSUE TITLE
Certain controls throw a static resource holder exception

### ISSUE THREAD

**Issue by Zohar127** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#1340]
I'm using Material Design in XAML on a Class Library project that is a plugin for AutoCAD. I've followed the guide and read other issues here from a user doing something similar in Revit. I ran into the same issues the Revit user had about loading the libraries and was able to mostly solve them.

First off, I'm "kickstarting" the assemblies by adding the following code to my MainWindow constructor in front of InitializeComponent:

`ColorZoneAssist.SetMode(new GroupBox(), ColorZoneMode.Accent);`
`Hue hue = new Hue("name", System.Windows.Media.Color.FromArgb(1, 2, 3, 4), System.Windows.Media.Color.FromArgb(1, 5, 6, 7));`

I need to kickstart because AutoCAD won't really allow the material design assemblies to run unless I create these dummy statements.

On my MainWindow I'm referencing the assemblies as per the quick start guide and have added all of the relevant lines to the window itself for foreground, textsize, etc...

Now here's my issue; Certain controls work, like Radio Button and Slider. They take on the nice themes and function as designed. Another one that works is `<materialdesign:card />` .

Weirdly, TextBoxes, Expanders, and `<materialdesign:PopupBox/>`  among other controls don't work. I call them basically like this:
`<Expander
            Style="{StaticResource MaterialDesignExpander}"
            Header="Hello I am an expander"
            />`

They throw a static resource holder exception. I've tried creating a resource dictionary right in the control for the specific themes like `<ResourceDictionary Source="pack://application:,,,/MaterialDesignThemes.Wpf;component/Themes/MaterialDesignTheme.Expander.xaml" />` but it gets me nowhere. Looks good in the designer but in run-time it causes that static resource holder exception.

I poked around inside the MaterialDesignTheme.Expander.xaml file to see if there's anything weird going on in there that might be causing these problems, but nothing stands out.

I assumed I needed to add another "kickstarter" to my MainWindow but I don't know what else I'd add, I have both assemblies and they are both copied over.

I'm on .net 4.6.1 and my version of VS is 2015 Professional. Is it possible that I need the latest version of Visual Studio and support for C# 7+? I noticed that I am unable to build the MaterialDesignForXaml toolkit on my machine because a lot of the actual code behind all of it is written with C# features that I'm pretty sure aren't supported on my older version, and something like an expander or popupbox seems like it would have some code logic behind it that a slider or radio box might not need.

**Comment by Zohar127** [comment]
I took the time to run through each control in MaterialDesignThemes and the following controls will crash my application. All other's work:


materialDesign:Chip
Expander
PopupBox
PasswordBox
RichTextBox
TextBox
TimePicker

What do these controls have in common that might be causing me an issue?

**Comment by jespersh** [comment]
Maybe a different method of "kick-starting" (because I don't know what you mean or working with), 
Action<Type> noop = _ => {};
var dummy = typeof(AbcDll.AnyClass);
noop(dummy);

Other methods here https://stackoverflow.com/questions/15816769/dependent-dll-is-not-getting-copied-to-the-build-output-folder-in-visual-studio

**Comment by Zohar127** [comment]
Thanks jespersh. I am "kickstarting" because AutoCAD is finnicky about loading outside .dlls. I can load an extension to run in AutoCAD using a built-in command (NETLOAD) but sometimes it's picky about what it allows to run as a reference. Usually you have to use some code to activate those other .dlls, like I was in the original post.

I tried your suggestion but unfortunately it's still giving me the same problem. However, I've really narrowed it down to not being a resource problem, at least not an obvious one.

I did

`        private static void Dummy1()
        {
            Action<Type> noop = _ => { };
            var dummy = typeof(MaterialDesignColors.Swatch);
            noop(dummy);
        }`

and

`        private static void Dummy2()
        {
            Action<Type> noop = _ => { };
            var dummy = typeof(MaterialDesignThemes.Wpf.PopupBox);
            noop(dummy);
        }`


I then added a PopUpBox to my xaml (looks good in designer) and ran the program and it crashed.  I removed the PopUpBox and ran the program and it worked again.

Kind of leading me back around to there being something specific about that small but essential handful of controls I listed out in my last post.

There's gotta be a solution! I'm dying to use this.

**Comment by Zohar127** [comment]
OK I am fairly positive I've tracked down exactly what is causing the static resource holder exception, and it's PackIcon.

All of the controls I listed above have PackIcon by default in their themes, SO I decided to change a previously working button to the style "MaterialDesignIconButton" and gave it an icon....

`<Button
                    Grid.Column="1"
                    Style="{StaticResource MaterialDesignIconButton}"
                    ToolTip="MaterialDesignIconButton">
                    <materialDesign:PackIcon
                        Kind="Play" />`

And it threw the exception. All of the controls which I have verified working don't have PackIcons in their default styles.

I've tried a few different methods to force the PackIcons to load but I'm not having any success, so at this point, I have to ask if anyone has any ideas on how I can force the PackIcon resources to load.

For reference...

My Resource Dictionary:
`<ResourceDictionary
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="clr-namespace:MatXAMLTest">

    <ResourceDictionary.MergedDictionaries>
        <ResourceDictionary
            Source="pack://application:,,,/MaterialDesignThemes.Wpf;component/Themes/MaterialDesignTheme.Light.xaml" />
        <ResourceDictionary
            Source="pack://application:,,,/MaterialDesignThemes.Wpf;component/Themes/MaterialDesignTheme.Defaults.xaml" />
        <ResourceDictionary
            Source="pack://application:,,,/MaterialDesignColors;component/Themes/Recommended/Primary/MaterialDesignColor.Blue.xaml" />
        <ResourceDictionary
            Source="pack://application:,,,/MaterialDesignColors;component/Themes/Recommended/Accent/MaterialDesignColor.Orange.xaml" />
    </ResourceDictionary.MergedDictionaries>
</ResourceDictionary>`

My MainWindow:
`<Window
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:local="clr-namespace:MatXAMLTest"
    xmlns:materialDesign="http://materialdesigninxaml.net/winfx/xaml/themes"
    xmlns:ControlzEx="clr-namespace:ControlzEx;assembly=MaterialDesignThemes.Wpf"
    x:Class="MatXAMLTest.MainWindow"
    mc:Ignorable="d"
    Title="MainWindow"
    TextElement.Foreground="{DynamicResource MaterialDesignBody}"
    TextElement.FontWeight="Medium"
    TextElement.FontSize="14"
    TextOptions.TextFormattingMode="Ideal"
    TextOptions.TextRenderingMode="Auto"
    Background="{DynamicResource MaterialDesignPaper}" FontFamily="pack://application:,,,/MaterialDesignThemes.Wpf;component/Resources/Roboto/#Roboto"
    Height="1200"
    Width="800"
    x:Name="page">`

```
   <!--Create References to styles-->
    <Window.Resources>
        <ResourceDictionary>
            <ResourceDictionary.MergedDictionaries>
                <ResourceDictionary
                    Source="pack://application:,,,/MatXAMLTest;component/MaterialDesign.xaml" />
            </ResourceDictionary.MergedDictionaries>
        </ResourceDictionary>
    </Window.Resources>

    <Grid>
        <Grid.RowDefinitions>
            <RowDefinition />
            <RowDefinition />
        </Grid.RowDefinitions>

        <materialDesign:ColorZone
            Grid.Row="0"
            Mode="Accent">
            <Grid>
                <Grid.ColumnDefinitions>
                    <ColumnDefinition />
                    <ColumnDefinition />
                </Grid.ColumnDefinitions>

                <Button
                    Margin="10"
                    Width="100"
                    Grid.Column="0"
                    Content="Hello" />
                <Button
                    Grid.Column="1"
                    Style="{StaticResource MaterialDesignIconButton}"
                    ToolTip="MaterialDesignIconButton">
                    <materialDesign:PackIcon
                        Kind="Play" />
                </Button>

            </Grid>
        </materialDesign:ColorZone>
    </Grid>
</Window>
```

And the various ways I tried to force-load PackIcons in my MainWindow.xaml.cs (none of which worked):

`        public MainWindow()
        {
            var icon = new PackIcon
            {
                Kind = PackIconKind.SmileyHappy,
                Foreground = System.Windows.Media.Brushes.Red
            };
            InitializeComponent();
        }`

`        private static void Dummy3()
        {
            Action<Type> noop = _ => { };
            var dummy = typeof(MaterialDesignThemes.Wpf.PackIcon);
            noop(dummy);
        }`

Any ideas?

**Comment by Keboo** [comment]
Perhaps related to #1327

**Comment by Zohar127** [comment]
@Keboo Possibly. Does that mean I'm kind of boned until your changes get merged with the master and the package gets updated? Is there some kind of way I can try to dummy load PackIcon? (I'm 100% new to git so forgive my ignorance).

**Comment by Keboo** [comment]
@Zohar127 currently the only option would be to check out that [branch](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/tree/fixPackIcon) and compile the library from source. I do have a backlog of pull requests I need to review and get merged, so I am planning on getting to that in the next week or so. I need to go back over my PR, and check a few things since the "fix" is a breaking API change.

**Comment by Zohar127** [comment]
@Keboo Ah thanks. In the meantime I guess I'll stick with using custom styles based on the MDIX controls with the PackIcon components removed as a workaround. I can't compile that branch because I don't have VS2017 or higher.

Bummed, but excited to continue playing with MDIX.

**Comment by Keboo** [comment]
I believe this is fixed in the latest 3.0.0 preview NuGet package. If there are still issues let me know.

**Comment by Zohar127** [comment]
I'll try it out. Thanks!
